backend/src/role.rs

```diff
@@ -0,0 +1,88 @@
+use crate::database::models::RoleUpdateInput;
+use crate::database::{
+    models::{Campaign, OrganisationUser, Role, User},
+    schema::AdminLevel,
+    Database,
+};
+use rocket::{
+    form::Form,
+    get, put,
+    serde::{json::Json, Serialize},
+};
+
+#[derive(Serialize)]
+pub enum RoleError {
+    RoleUpdateFailure,
+    RoleNotFound,
+    CampaignNotFound,
+    Unauthorized,
+}
+
+#[derive(Serialize)]
+pub struct RoleResponse {
+    name: String,
+    description: Option<String>,
+    min_available: i32,
+    max_available: i32,
+}
+
+#[get("/<role_id>")]
+pub async fn get_role(
+    role_id: i32,
+    _user: User,
+    db: Database,
+) -> Result<Json<RoleResponse>, Json<RoleError>> {
+    let res: Option<Role> = db.run(move |conn| Role::get_from_id(&conn, role_id)).await;
+
+    match res {
+        Some(role_) => Ok(Json(RoleResponse {
+            name: role_.name,
+            description: role_.description,
+            min_available: role_.min_available,
+            max_available: role_.max_available,
+        })),
+        None => Err(Json(RoleError::RoleNotFound)),
+    }
+}
+
+#[put("/<role_id>", data = "<role_update>")]
+pub async fn update_role(
+    role_id: i32,
+    role_update: Form<RoleUpdateInput>,
+    _user: User,
```

Comment:
this should be `user` - we only denote variables with `_` if they are unused, but we use `user` in this fn